packages/components/src/tools-panel/test/__snapshots__/index.js.snap

```diff
@@ -5,6 +5,7 @@ exports[`ToolsPanel first and last panel items should apply first/last classes t
   display: grid;
   gap: calc( 4px * 3 );
   grid-template-columns: repeat( 2, 1fr );
+  grid-template-columns: repeat( 2, minmax(0, 1fr) );
```

Comment:
It would be interesting to see why there's two `grid-template-columns` rules being applied here, as it may be the symptom of an edge case that we haven't considered yet

Reply:
Thanks for catching this one 👍 

The first `grid-template-columns` rule comes from the `Grid` component. It [generates](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/grid/hook.ts#L55-L66) this rule if it has a `columns` value. This `Grid` prop gets [defaulted](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/grid/hook.ts#L29) to `2` and is typed only to accept a `number` currently.

The grid template columns defined by the grid component aren't sufficient for the `ToolsPanel` use case so we add the one with `minmax()` in this PR. 

We originally switched the `ToolsPanel` to [render via a `Grid`](https://github.com/WordPress/gutenberg/pull/35621) to get some of this layout for free. It seems we'll be benefiting from this less now.

Do you think it would be worthwhile to provide a prop on the `Grid` component to optionally enforce equal width columns?

Reply:
Thank you for the detailed context, as always!

Your explanation and suggestions make sense, but I'd like to discuss these changes with @mirka in a broader sense, since some of `Grid`'s prop are mirroring the exact CSS property (e.g. justify, align, columnGap, rowGap, templateColumns, templateRows...) while other are not (e.g gap, isInline, alignment...). I'd like to first make some clarity on the direction we want to take with `Grid` (and similarly with `Flex`), before adding a new prop

Feel free to merge this PR, if the rest of the feedback is addressed!